### PR TITLE
[ADD] event_sale_reservation: sell events before they exist

### PR DESCRIPTION
**Pull request by yajo**
This module extends the functionality of event_sale to support selling
reservations of events that still don't exist and to allow you to schedule the
creation of events based on how many reservations already exist.

@Tecnativa TT27664

# Pictures

screenshot-2021-02-01-11-51-25

screenshot-2021-02-01-11-56-34

screenshot-2021-02-01-12-00-00

screenshot-2021-02-01-12-04-36

### CODE REVIEW

File: event_sale_reservation/README.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+If you want to convert those reservations into real event registrations:
+
+#. Go to the quotation you just created (if you are not there yet).
+#. Click on *Register in event*.
+#. In the wizard you see, set the *Event* and *Event Ticket* for all the order
+   lines you want to convert into registrations.
+#. If there is any line you still don't want to convert, remove it from the
+   wizard.
+#. Click on *Next*.
+#. A new wizard will appear, where you will be able to specify the name, email
```
Comment (chienandalu) [inline event_sale_reservation/README.rst:104]: Doing the functional test I got stuck here, as the second wizard didn't pop up but the line got the event and ticket info but the registrations weren't made.

I guess has something to do with the `web_ir_actions_act_multi` action

Comment (yajo) [reply event_sale_reservation/README.rst:104]: I pushed a fix, thanks. Could you try again?

File: event_sale_reservation/wizards/registration_editor.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            for sol in order.order_line:
+                if not sol.product_id.event_reservation_ok:
+                    continue
+                result["event_registration_ids"] += [
+                    (
+                        0,
+                        0,
+                        {
+                            "sale_order_line_id": sol.id,
+                            "event_reservation_type_id": sol.event_reservation_type_id.id,
```
Comment (chienandalu) [inline event_sale_reservation/wizards/registration_editor.py]: Too long

**Review by chienandalu — COMMENTED** [review]
Now runbot seems to be down :sa: 

For the moment...:

File: event_sale_reservation/wizards/registration_editor.py (hunk 2 above)
Comment (yajo) [reply event_sale_reservation/wizards/registration_editor.py]: Thanks, fixed.

**Comment by yajo** [comment]
All fixed. New feature:

imagen

**Review by chienandalu — COMMENTED** [review]
Tests failing

**Comment by yajo** [comment]
Yes, thanks. It should be OK now.

**Review by chienandalu — CHANGES_REQUESTED** [review]
When I click on the sale smart button to the event registrations I get this error:
` Uncaught Error: NameError: name 'id' is not defined` You can try in this runbot (It it's still alive: http://3465580-203-b8607e.runbot2-3.odoo-community.org/web?debug=assets#id=23&action=398&model=sale.order&view_type=form&menu_id=253)

**Comment by pedrobaeza** [comment]
@Yajo that's why I told you about not using `id`. Please change it to methods returning the corresponding action previously read and completed through code.

**Review by chienandalu — APPROVED** [review]
Tested functionaly :slightly_smiling_face: :+1:

**Comment by yajo** [comment]
@pedrobaeza fixed

**Comment by pedrobaeza** [comment]
@sergio-teruel is it OK for you?

**Comment by sergio-teruel** [comment]
@pedrobaeza Yes.

**Pull request merged**